File: src/dotnet/Src/ButterflyNet.Parser/Scopes/PreformattedLineScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class PreformattedLineScope : BlockScope {
		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenPreformattedLine();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.ClosePreformattedLine();
		}
	}
}

File: src/js/src/scopes/PreformattedLineScope.js
function PreformattedLineScope() {
	this.type = ScopeTypeCache.preformattedLine;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openPreformattedLine();
	};
	this.close = function(analyzer) {
		analyzer.closePreformattedLine();
	};
}

extend(Scope, PreformattedLineScope);